datalad/tests/test_annexrepo.py

```diff
@@ -367,6 +367,49 @@ def test_AnnexRepo_get_file_backend(src, dst):
         assert_raises(CommandNotAvailableError, ar.migrate_backend,
                       'test-annex.dat', backend='SHA1')
 
+
+@with_tempfile
+def test_AnnexRepo_always_commit(path):
+
+    repo = AnnexRepo(path)
+    file1 = get_most_obscure_supported_name() + "_1"
+    file2 = get_most_obscure_supported_name() + "_2"
+    with open(opj(path, file1), 'w') as f:
+        f.write("First file.")
+    with open(opj(path, file2), 'w') as f:
+        f.write("Second file.")
+
+    repo.always_commit = True
+    repo.annex_add(file1)
+
+    # Now git-annex log should show the addition:
+    out, err = repo._run_annex_command('log')
+    out_list = out.rstrip(os.linesep).split(os.linesep)
+    assert_equal(len(out_list), 1)
+    assert_in(file1, out_list[0])
+
+    repo.always_commit = False
+    repo.annex_add(file2)
+
+    repo.always_commit = True
+
+    # Still one commit only in log:
+    out, err = repo._run_annex_command('log')
+    out_list = out.rstrip(os.linesep).split(os.linesep)
+    assert_equal(len(out_list), 2, "Output:\n%s" % out_list)
+    assert_in(file1, out_list[0])
+    assert_in("recording state in git", out_list[1])
+
+    # but 'git annex log' was called when always_commit was true again,
+    # so it committed the addition at the end. Calling it again should show
```

Comment:
so this comment is relevant already for right after running above 'log' -- move it there.
But that is why we should also check with regular git that it is still uncomitted before running annex log